syntax = "proto3";
package agent;

service IrisInference {
    rpc Inference (IrisInferenceRequest) returns (IrisInferenceResponse);
}

message IrisInferenceRequest {
    string id = 1;

    float sepal_length = 2;
    float sepal_width = 3;
    float petal_length = 4;
    float petal_width = 5;
}

message IrisInferenceResponse {
    string species = 1;
}

service MnistInference {
    rpc Inference (MnistInferenceRequest) returns (MnistInferenceResponse);
}

message MnistInferenceRequest {
    string id = 1;
    string file_path = 2;
}

message MnistInferenceResponse {
    string file_path = 1;
}

service FileInference {
    rpc Inference (FileInferenceRequest) returns (FileInferenceResponse);
}

message FileInferenceRequest {
    string file_path = 1;
    string ip = 2;
    float port = 3;
    float file_size = 4;
    bool is_remote = 5;
}

message FileInferenceResponse {
    string file_path = 1;
    string ip = 2;
    float port = 3;
    float file_size = 4;
    bool is_remote = 5;
}